File: array-manipulation/array-manipulation.js
function arrayManipulation(n, queries) {
  /* Variables */
  let myArray = [];
  let output = 0;
  let itt = 0;

  /* Array of zeros */
  for (let i = 0; i < n; i++) {
    myArray.push(0);
  }

  /* Queries */
  for (let i = 0; i < queries.length; i++) {
    myArray[queries[i][0] - 1] += queries[i][2];
    if (queries[i][1] != myArray.length) {
      myArray[queries[i][1]] -= queries[i][2];
    }
  }
  console.log(myArray);

  /* Maximum value */
  for (let j = 0; j < myArray.length; j++) {
    itt += myArray[j];
    if (itt > output) {
      output = itt;
    }
  }
  return output;
}

let test = arrayManipulation(10, [
  //a b  k 
  [1, 5, 3],
  [4, 8, 7],
  [6, 9, 1]]);

console.log(test);

File: array-manipulation/array-manipulation.py
#Code solution from https://sites.northwestern.edu/acids/2018/11/12/solution-hackerrank-array-manipulation/

def arrayManipulation(n, queries):
    arr = [0] * n

    for i in queries:
        arr[i[0] - 1] += i[2]
        if i[1] != len(arr):
            arr[i[1]] -= i[2]

    maxval = 0
    itt = 0
    print(arr)

    for q in arr:
        itt += q
        if itt > maxval:
            maxval = itt

    print(maxval)
    return maxval


arrayManipulation(10, [[1, 5, 3], [4, 8, 7], [6, 9, 1]])
